src/proto/properties.rs

```diff
@@ -0,0 +1,202 @@
+                "difficulty" => p.difficulty = value.parse().unwrap(),
+                "enable-query" => p.enable_query = value.parse().unwrap(),
+                "enable-rcon" => p.enable_rcon = value.parse().unwrap(),
+                "enable-command-block" => p.enable_command_block = value.parse().unwrap(),
+                "force-gamemode" => p.force_gamemode = value.parse().unwrap(),
+                "gamemode" => p.gamemode = value.parse().unwrap(),
+                "generate-structures" => p.generate_structures = value.parse().unwrap(),
+                "generator-settings" => p.generator_settings = value.to_string(),
+                "hardcore" => p.hardcore = value.parse().unwrap(),
+                "level-name" => p.level_name = value.to_string(),
+                "level-seed" => p.level_seed = value.to_string(),
+                "level-type" => p.level_type = value.to_string(),
+                "max-build-height" => p.max_build_height = value.parse().unwrap(),
+                "max-players" => p.max_players = value.parse().unwrap(),
+                "max-tick-time" => p.max_tick_time = value.parse().unwrap(),
+                "max-world-size" => p.max_world_size = value.parse().unwrap(),
+                "motd" => p.motd = value.to_string(),
+                "network-compression-threshold" => p.network_compression_threshold = value.parse().unwrap(),
+                "online-mode" => p.online_mode = value.parse().unwrap(),
+                "op-permission-level" => p.op_permission_level = value.parse().unwrap(),
+                "player-idle-timeout" => p.player_idle_timeout = value.parse().unwrap(),
+                "pvp" => p.pvp = value.parse().unwrap(),
+                "query.port" => p.query_port = value.parse().unwrap(),
+                "rcon.password" => p.rcon_password = value.to_string(),
+                "rcon.port" => p.rcon_port = value.parse().unwrap(),
+                "resource-pack" => p.resource_pack = value.to_string(),
+                "resource-pack-hash" => p.resource_pack_hash = value.to_string(),
+                "server-ip" => p.server_ip = value.to_string(),
+                "server-port" => p.server_port = value.parse().unwrap(),
+                "snooper-enabled" => p.snooper_enabled = value.parse().unwrap(),
+                "spawn-animals" => p.spawn_animals = value.parse().unwrap(),
+                "spawn-monsters" => p.spawn_monsters = value.parse().unwrap(),
+                "spawn-npcs" => p.spawn_npcs = value.parse().unwrap(),
+                "spawn-protection" => p.spawn_protection = value.parse().unwrap(),
+                "use-native-transport" => p.use_native_transport = value.parse().unwrap(),
+                "view-distance" => p.view_distance = value.parse().unwrap(),
+                "white-list" => p.white_list = value.parse().unwrap(),
+                prop => println!("Unknown property {}", prop),
```

Comment:
Actually... it may be good that it's a `println!`... We don't need to error out if a key is unknown, just log it to console. Right?